src/routing/approved.rs

```diff
@@ -55,6 +55,7 @@ pub(crate) struct Approved {
     relocate_state: Option<RelocateState>,
     msg_filter: MessageFilter,
     pub(super) event_tx: mpsc::UnboundedSender<Event>,
+    join_flag: bool,
```

Comment:
Maybe call it `joins_allowed`?

Reply:
yeah, sounds better.
will change to it if no other suggestions.